builtin/logical/pki/path_config_acme.go

```diff
@@ -226,9 +227,34 @@ func (b *backend) pathAcmeWrite(ctx context.Context, req *logical.Request, d *fr
 		config.EabPolicyName = eabPolicy.Name
 	}
 
+	// Validate Default Directory Behavior:
+	defaultDirectoryPolicyIsRole := false
+	switch {
+	case config.DefaultDirectoryPolicy == "forbid" || config.DefaultDirectoryPolicy == "sign-verbatim":
+	case strings.HasPrefix(config.DefaultDirectoryPolicy, "role:"):
+		defaultDirectoryPolicyIsRole = true
+
```

Comment:
this is a slight change in structure, but previously the role configured by default_role was not being validated when allowAnyRole was true (that is allowed_roles was "*")